src/ProjectNeon/Assets/Features/Effects/RepeatOnPrimaryResource.cs

```diff
@@ -0,0 +1,23 @@
+﻿using UnityEngine;
+using System.Collections;
+
+/**
+ * Apply an effect while there is enough Primary Resource
+ */
+public class RepeatOnPrimaryResource : Effect
```

Comment:
The name is unclear when reading the card in the editor. Maybe name this `RepeatUntilCannotAffordPrimaryResourceCost` or something that expresses the terminating condition.